crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -102,6 +102,36 @@ fn is_non_ascii_identifier_start(c: char) -> bool {
     is_xid_start(c)
 }
 
+/// Returns `true` if the character is part of a Python operator (like `+=` or `->`).
+const fn is_any_operator_token(c: char) -> bool {
+    matches!(
+        c,
+        '=' | '+'
+            | '*'
+            | '/'
+            | '%'
+            | '|'
+            | '^'
+            | '&'
+            | '-'
+            | '@'
+            | '!'
+            | ':'
+            | ';'
+            | '<'
+            | '>'
+            | ','
+            | '.'
+            | '('
+            | ')'
+            | '['
+            | ']'
+            | '{'
+            | '}'
+            | '~'
+    )
```

Comment:
These (and some others like `:`, `!`) are never ambiguous. We can return the first token right away when backward lexing.